File: packages/swift/Sources/CatalystSwift/Models/EpicleseWorld.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

/// A world/scene registered on a given platform.
public struct EpicleseWorld: Decodable, Sendable, Equatable, Hashable {
  public let id: String
  public let platformIdentifier: String
  public let platform: EpiclesePlatform?
  public let name: String
  public let author: EpicleseAuthor?
}

/// Wrapper for the `{"world": {...} | null}` response shape.
public struct EpicleseWorldWrapper: Decodable, Sendable {
  public let world: EpicleseWorld?
}

/// Wrapper for the `{"items": [...]}` response shape.
public struct EpicleseWorldsResult: Decodable, Sendable {
  public let items: [EpicleseWorld]
}

/// Request body for `POST /epiclese/v1/worlds`.
public struct EpicleseCreateWorldAuthorRequest: Encodable, Sendable {
  public let id: String?
  public let platformIdentifier: String
  public let platform: String
  public let name: String

  public init(id: String? = nil, platformIdentifier: String, platform: String, name: String) {
    self.id = id
    self.platformIdentifier = platformIdentifier
    self.platform = platform
    self.name = name
  }
}

/// Request body for `POST /epiclese/v1/worlds`.
public struct EpicleseCreateWorldRequest: Encodable, Sendable {
  public let id: String?
  public let platformIdentifier: String
  public let platform: String
  public let name: String
  public let author: EpicleseCreateWorldAuthorRequest?

  public init(
    id: String? = nil, platformIdentifier: String, platform: String, name: String,
    author: EpicleseCreateWorldAuthorRequest? = nil
  ) {
    self.id = id
    self.platformIdentifier = platformIdentifier
    self.platform = platform
    self.name = name
    self.author = author
  }
}

File: packages/kotlin/src/main/kotlin/com/natsuneko/catalyst/models/EpicleseWorld.kt
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

package com.natsuneko.catalyst.models

import kotlinx.serialization.Serializable

/**
 * A world/scene registered on a given platform
 */
@Serializable
data class EpicleseWorld(
    val id: String,
    val platformIdentifier: String,
    val platform: EpiclesePlatform? = null,
    val name: String,
    val author: EpicleseAuthor? = null
)

/**
 * Wrapper for the `{"world": {...} | null}` response shape
 */
@Serializable
internal data class EpicleseWorldWrapper(val world: EpicleseWorld? = null)

/**
 * Wrapper for the `{"items": [...]}` response shape
 */
@Serializable
internal data class EpicleseWorldsResult(val items: List<EpicleseWorld>)
